pom.xml

```diff
@@ -891,20 +892,19 @@ from system library in Java 11+ -->
                     <plugin>
                         <groupId>org.apache.maven.plugins</groupId>
                         <artifactId>maven-checkstyle-plugin</artifactId>
-                        <version>3.1.2</version>
+                        <version>3.4.0</version>
                         <dependencies>
                             <dependency>
                                 <groupId>com.puppycrawl.tools</groupId>
                                 <artifactId>checkstyle</artifactId>
-                                <version>10.3.2</version>
+                                <version>10.17.0</version>
                             </dependency>
                         </dependencies>
                         <executions>
                             <execution>
                                 <id>checkstyle-check</id>
                                 <phase>validate</phase>
                                 <configuration>
-                                    <encoding>UTF-8</encoding>
```

Comment:
I think this should be changed to `<inputEncoding>` instead of being removed.

Reply:
See https://maven.apache.org/plugins/maven-checkstyle-plugin/check-mojo.html#inputEncoding. So you are right, `<inputEncoding>` would work, but it is not necessary because we already set the correct default encoding `<project.build.sourceEncoding>`.

Reply:
Right, thanks for pointing this out.